src/test/java/build/buildfarm/BuildFarmWorkerTest.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm;
+
+import build.buildfarm.v1test.WorkerConfig;
+import build.buildfarm.worker.Worker;
+import javax.naming.ConfigurationException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BuildFarmWorkerTest {
+  @Test(expected = ConfigurationException.class)
+  public void missingWorkerRoot() throws ConfigurationException {
+    WorkerConfig.Builder builder = WorkerConfig.newBuilder();
+    new Worker(builder.build());
+  }
+
+  @Test(expected = ConfigurationException.class)
+  public void missingCasCacheDirectory() throws ConfigurationException {
+    WorkerConfig.Builder builder = WorkerConfig.newBuilder();
+    builder.setRoot("/");
+    new Worker(builder.build());
+  }
+}
+
```

Comment:
Trailing empty line

Reply:
Would like to keep that to be in git's good graces.

Reply:
Trailing newline is not the same thing as trailing empty line, but w/e